tests_e2e/tests/agent_status/agent_status.py

```diff
@@ -0,0 +1,202 @@
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = self._context.create_ssh_client()
+
+    def validate_instance_view_vmagent_status(self, instance_view: VirtualMachineInstanceView):
+        status = instance_view.vm_agent.statuses[0]
+
+        # Validate message field
+        message = status.message
+        if message is None:
+            raise RetryableAgentStatusException("Instance view is missing an agent status message")
+        elif 'unresponsive' in message:
+            raise RetryableAgentStatusException("Instance view shows unresponsive agent")
+
+        # Validate display status field
+        display_status = status.display_status
+        if display_status is None:
+            raise RetryableAgentStatusException("Instance view is missing an agent display status")
+        elif 'Not Ready' in display_status:
+            raise RetryableAgentStatusException("Instance view shows agent status is not ready")
+
+        # Validate time field
+        time = status.time
+        if time is None:
+            raise RetryableAgentStatusException("Instance view is missing an agent status timestamp")
+
+    def validate_instance_view_vmagent(self, instance_view: VirtualMachineInstanceView):
+        # Validate vm_agent_version field
+        vm_agent_version = instance_view.vm_agent.vm_agent_version
+        if vm_agent_version is None:
+            raise RetryableAgentStatusException("Instance view is missing agent version")
+        elif 'Unknown' in vm_agent_version:
+            raise RetryableAgentStatusException("Instance view shows agent version is unknown")
+
+        # Validate statuses field
+        statuses = instance_view.vm_agent.statuses
+        if statuses is None:
+            raise RetryableAgentStatusException("Instance view is missing agent statuses")
+        elif len(statuses) < 1:
+            raise RetryableAgentStatusException("Instance view is missing an agent status entry")
+        else:
+            self.validate_instance_view_vmagent_status(instance_view=instance_view)
+
+    def validate_instance_view(self, instance_view: VirtualMachineInstanceView):
+        """
+        Checks that instance view has vm_agent.statuses and vm_agent.vm_agent_version properties which report the Guest
+        Agent as running and Ready:
+
+        "vm_agent": {
+            "extension_handlers": [],
+            "vm_agent_version": "9.9.9.9",
+            "statuses": [
+                {
+                    "level": "Info",
+                    "time": "<class 'datetime.datetime'>",
+                    "message": "Guest Agent is running",
+                    "code": "ProvisioningState/succeeded",
+                    "display_status": "Ready"
+                }
+            ]
+        }
+        """
+        if instance_view.vm_agent is None:
+            raise RetryableAgentStatusException("Instance view is missing vm agent")
```

Comment:
If you want to check that the property is present you should do '"vm_agent" in instance_view". The dot operator would raise an exception if the property is missing.

Same applies for the other checks on properties

Reply:
I updated this to have a more accurate error message "{property} in instance view is None" and also typed my instance_view variables.

I'm using the dot operator because the Model azure.mgmt.compute.models has classes for InstanceViewStatus and VirtualMachineAgentInstanceView. All of the properties I'm checking are attributes of these classes and initialized to None:
```
class InstanceViewStatus(msrest.serialization.Model):
    def __init__(
        self,
        *,
        code: Optional[str] = None,
        level: Optional[Union[str, "StatusLevelTypes"]] = None,
        **display_status: Optional[str] = None,
        message: Optional[str] = None,
        time: Optional[datetime.datetime] = None,**
        **kwargs
    ):
        super(InstanceViewStatus, self).__init__(**kwargs)
        self.code = code
        self.level = level
        self.display_status = display_status
        self.message = message
        self.time = time
```
```
class VirtualMachineAgentInstanceView(msrest.serialization.Model):
    def __init__(
        self,
        *,
        vm_agent_version: Optional[str] = None,
        extension_handlers: Optional[List["VirtualMachineExtensionHandlerInstanceView"]] = None,
        statuses: Optional[List["InstanceViewStatus"]] = None,
        **kwargs
    ):
        super(VirtualMachineAgentInstanceView, self).__init__(**kwargs)
        self.vm_agent_version = vm_agent_version
        self.extension_handlers = extension_handlers
        self.statuses = statuses
```

Reply:
cool. could you add that as a comment?